hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -3987,7 +3987,8 @@ public Object performTask() throws IOException {
               + " deleted and the block removal is delayed");
         }
         INodeFile iFile = ((INode)storedBlock.getBlockCollection()).asFile();
-        if (inodeIdentifier==null) {
+        INode inode = EntityManager.find(INode.Finder.ByINodeIdFTIS, inodeIdentifier.getInodeId());
```

Comment:
I think we should add another check if inodeIdentifier is not null before calling find to avoid NPE in inodeIdentifier.getInodeId()

Reply:
fixed